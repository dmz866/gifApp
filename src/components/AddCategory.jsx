import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategory = ({ addCategory }) => {
	const [inputValue, setInputValue] = useState('');
	const onInputChange = (e) => setInputValue(e.target.value);
	const onSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim()) {
			addCategory(inputValue);
		}

		setInputValue('');
	};

	return (
		<form onSubmit={onSubmit} aria-label='form'>
			<input type='text' placeholder='Buscar gifs' value={inputValue} onChange={onInputChange} />
		</form>
	)
};

AddCategory.propTypes = {
	addCategory: PropTypes.func.required
};
