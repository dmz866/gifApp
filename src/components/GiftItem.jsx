import PropTypes from 'prop-types';

export const GiftItem = ({ titulo, url }) => {
	return (
		<div className="card">
			<img src={url} alt={titulo} />
			<p>{titulo}</p>
		</div>
	)
};

GiftItem.propTypes = {
	titulo: PropTypes.string.required,
	url: PropTypes.string.required
};