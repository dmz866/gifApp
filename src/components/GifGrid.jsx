import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GiftItem } from './GiftItem';

export const GifGrid = ({ category }) => {
	const { images, loading } = useFetchGifs(category);

	return (
		<div>
			<h3>{category}</h3>
			{loading && <h2>Cargando...</h2>}
			<div className="card-grid">
				{images.map((image) => (
					<GiftItem key={image.id} {...image} />
				))}
			</div>
		</div>
	)
};

GifGrid.propTypes = {
	category: PropTypes.string.required
};

