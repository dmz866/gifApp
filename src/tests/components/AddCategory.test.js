import { fireEvent, render, screen } from "@testing-library/react";
import { AddCategory } from "../../components";

describe('Test AddCategory', () => {

	test('debe cambiar el valor de la caja de texto', () => {
		render(<AddCategory addCategory={() => { }} />);
		const input = screen.getByRole('textbox');

		fireEvent.input(input, { target: { value: 'ValueTest' } });

		expect(input.value).toBe('ValueTest');
	});

	test('debe llamar a addCategory si el input tiene un valor', () => {
		const inputValue = 'valueTest';
		const addCategory = jest.fn();

		render(<AddCategory addCategory={addCategory} />);

		const input = screen.getByRole('textbox');
		const form = screen.getByRole('form');

		fireEvent.input(input, { target: { value: inputValue } });
		fireEvent.submit(form);

		expect(input.value).toBe('');
		expect(addCategory).toBeCalled();
		expect(addCategory).toBeCalledTimes(1);
		expect(addCategory).toBeCalledWith(inputValue);
	});

	test('no debe llamar a addCategory si el input esta vacio', () => {
		const addCategory = jest.fn();

		render(<AddCategory addCategory={addCategory} />);

		const form = screen.getByRole('form');
		fireEvent.submit(form);

		expect(addCategory).not.toBeCalled();
	});
});