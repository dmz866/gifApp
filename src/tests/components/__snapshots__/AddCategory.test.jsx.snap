// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddCategory debe cambiar el valor de la caja de texto 1`] = `
<div>
  <form>
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
    <button>
      +
    </button>
  </form>
</div>
`;

exports[`Test GiftItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="tituloTest"
      src="urlTest"
    />
    <p>
      tituloTest
    </p>
  </div>
</div>
`;
